app.rb

```diff
@@ -1,7 +1,30 @@
 require 'sinatra/base'
+require_relative './lib/player.rb'
+require_relative './lib/game.rb'
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+  set :public_folder, 'public'
+  get '/' do
+    erb :index
+  end
+
+  post '/names' do
+    @game = Game.create(Player.new(params[:name]))
+    redirect '/play'
+  end
+
+  before do
+    @game = Game.instance
+  end
+
+  get '/play' do
+    erb :play
+  end
+
+  post '/game' do
+    @game.player.choice = params[:action].downcase.to_sym
+    @result = @game.calculate_result.upcase
+    erb :result
   end
```

Comment:
Simple, to the point, and readable. Good code. Only part I would change is the return you do in `@game.calculate_result` to be upcase, so you just call `@game.calculate_result` here.